src/Components/Web.JS/src/Rendering/JSRootComponents.ts

```diff
@@ -7,7 +7,7 @@ const pendingRootComponentContainerNamePrefix = '__bl-dynamic-root:';
 const pendingRootComponentContainers = new Map<string, Element>();
 let nextPendingDynamicRootComponentIdentifier = 0;
 
-type ComponentParameters = object | null | undefined;
+type ComponentParameters = Record<string, unknown>
```

Comment:
Is this definitely correct? What was wrong with it before? The previous definition was more specific about the potential for null/undefined which looks more relevant.

Reply:
Yes. `object` is the ban type.
Reference -> https://github.com/typescript-eslint/typescript-eslint/blob/v4.32.0/packages/eslint-plugin/docs/rules/ban-types.md

Reply:
`type ComponentParameters = {
    [x: string]: unknown;
}`
This is how the type has been inferred. The `unknown` can be `null` or `undefined`! I know it's not that verbose, if you think it might confuse people we can ignore this warning!

Reply:
I know the values inside the record can be null/undefined. The thing I'm talking about is that `ComponentParameters` *itself* can be null/undefined.